lib/ansible/modules/network/checkpoint/cp_network.py

```diff
@@ -0,0 +1,179 @@
+options:
+  name:
+    description:
+      - Object name.
+    type: str
+  subnet:
+    description:
+      - IPv4 or IPv6 network address. If both addresses are required use subnet4 and subnet6 fields explicitly.
+    type: str
+  subnet4:
+    description:
+      - IPv4 network address.
+    type: str
+  subnet6:
+    description:
+      - IPv6 network address.
+    type: str
+  mask_length:
+    description:
+      - IPv4 or IPv6 network mask length. If both masks are required use mask-length4 and mask-length6 fields explicitly. Instead of IPv4 mask length it is possible to specify IPv4 mask itself in subnet-mask field.
+    type: int
+  mask_length4:
+    description:
+      - IPv4 network mask length.
+    type: int
+  mask_length6:
+    description:
+      - IPv6 network mask length.
+    type: int
+  subnet_mask:
+    description:
+      - IPv4 network mask.
+    type: str
+  nat_settings:
+    description:
+      - NAT settings.
+    type: dict
+  tags:
+    description:
+      - Collection of tag identifiers.
+    type: list
+  broadcast:
+    description:
+      - Allow broadcast address inclusion.
+    type: str
+    choices: ['disallow', 'allow']
+  color:
+    description:
+      - Color of the object. Should be one of existing colors.
+    type: str
+    choices: ['aquamarine', 'black', 'blue', 'crete blue', 'burlywood', 'cyan', 'dark green', 'khaki', 'orchid', 'dark orange', 'dark sea green', 'pink', 'turquoise', 'dark blue', 'firebrick', 'brown', 'forest green', 'gold', 'dark gold', 'gray', 'dark gray', 'light green', 'lemon chiffon', 'coral', 'sea green', 'sky blue', 'magenta', 'purple', 'slate blue', 'violet red', 'navy blue', 'olive', 'orange', 'red', 'sienna', 'yellow']
+  comments:
+    description:
+      - Comments string.
+    type: str
+  details_level:
+    description:
+      - The level of detail for some of the fields in the response can vary from showing only the UID value of the object to a fully detailed representation of the object.
+    type: str
+    choices: ['uid', 'standard', 'full']
+  groups:
+    description:
+      - Collection of group identifiers.
+    type: list
+  ignore_warnings:
+    description:
+      - Apply changes ignoring warnings.
+    type: bool
+  ignore_errors:
+    description:
+      - Apply changes ignoring errors. You won't be able to publish such a changes. If ignore-warnings flag was omitted - warnings will also be ignored.
+    type: bool
+  uid:
+    description:
+      - Object unique identifier.
+    type: str
+  new_name:
+    description:
+      - New name of the object.
+    type: str
+extends_documentation_fragment: checkpoint_objects
+"""
+
+EXAMPLES = """
+- name: Add network object
+  cp_network:
+    name: New Network 3
+    nat_settings:
+      auto-rule: true
+      hide-behind: ip-address
+      install-on: All
+      ip-address: 192.0.2.1
+      method: static
+    state: present
+    subnet: 192.0.2.1
+    subnet_mask: 255.255.255.0
+"""
```

Comment:
@chkp-orso plz include example for `set` and `delete` as well.

Reply:
@justjais But I saw many vendors (like fortimanager) write only add example.
btw Riki also didn't write examples for add,set and delete

Reply:
@chkp-orso, It's better to add straightforward examples as user's using respective module can directly refer the examples given n modules and use the module accordingly as per need, without going and referring module documentation. So, please add respective examples as that'll make module user friendly. Also, please don't follow `fortinet/fortimanager/fortios` modules as they have not followed many of Ansible std. earlier and now they are planning to do loads of rework.

Reply:
amm OK I will add it

Reply:
@justjais ok I will add those examples